src/cmds/fs/mount.c

```diff
@@ -83,53 +84,46 @@ static void show_mount_list(void) {
 
 int main(int argc, char **argv) {
 	int opt;
-	int opt_cnt;
+	bool fs_type_passed = false;
 	char *dev, *dir;
 	char *fs_type;
 
-	opt_cnt = 0;
-	fs_type = 0;
+	if (argc == 1) {
+		show_mount_list();
+		return 0;
+	}
+
+	fs_type = NULL;
 	getopt_init();
 	while (-1 != (opt = getopt(argc, argv, "ht:"))) {
 		switch (opt) {
-		case 't':
-			opt_cnt++;
-			break;
-		case '?':
-			break;
-		case 'h':
-			print_usage();
-			/* FALLTHROUGH */
-		default:
-			return 0;
-		}
-	}
-
-	if (argc > 2) {
-		dev = argv[argc - 2];
-		dir = argv[argc - 1];
-		if (argc > 3) {
-			if (opt_cnt) {
-				fs_type = argv[argc - 3];
-			} else {
+			case 't':
+				fs_type_passed = true;
```

Comment:
Why do not set `fs_type = optarg` here? And use `!fs_type` instead of `!fs_type_passed` below?

Reply:
Done

Reply:
Thanks